site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+
+### `hostnames` field
+
+GAMMA implementations SHOULD NOT infer any functionality from the `hostnames` field on `HTTPRoute` due to current under-specification and reserved potential for future usage or API changes. For the use case of filtering incoming traffic from selected hostnames, it is recommended to guide users toward configuring [`HTTPHeaderMatch`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPHeaderMatch) rules for the [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header. Functionality to be explored in future GEPs may include supporting concurrent usage of an `HTTPRoute` traffic configuration for multiple North/South `Gateways` and East/West mesh use cases or redirection of egress traffic to an in-cluster `Service`.
+
+### Limiting graph depth, preventing conflicts and cycles
+
+A `Service` may only be specified as a `parentRef` of an `HTTPRoute` if it is:
+
+* NOT already specified as a `parentRef` of another `HTTPRoute` [in the same namespace]
+    * This requirement may be relaxed later to follow the existing [precedence rules](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule), but is intended to simplify initial implementation.
+    * The namespace colocation requirement effectively forecloses using this strategy to delegate route management, but there is service mesh [user interest](https://github.com/istio/istio/issues/22997) in merging separately-defined routes together to manage large configurations, multiple versions or canary deployments for a single mesh service.
+* NOT already specified as a `backendRef` of any _other_ `HTTPRoute` with a `Service` `parentRef`
+    * A `Service` MAY still be listed as a `backendRef` on the _same_ `HTTPRoute` where it is specified as a `parentRef`, e.g. to only split off some subset of traffic from the default backend endpoints.
+    * This could be computationally expensive to validate.
+
+This should be enforced by controller validation, which should set an `Accepted: { status: False }` condition with a corresponding new `ParentRefConflict` `RouteConditionReason`.
+
+The logic described here is a bit verbose, but could be validated with conformance tests, and is intended to allow an `HTTPRoute` configuring mesh traffic rules to additionally specify a `Gateway` `parentRef` (or `HTTPRoute` `parentRef` if [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085) moves forward) to apply the same routing rules when exposing a mesh service outside the cluster.
```

Comment:
Would definitely appreciate 👀 on this section to make sure the scope of these rules feels reasonable and sufficient to cover the goal of allowing a Service to both have E/W mesh routing rules applied and be accessible through a Gateway concurrently.

Reply:
This sounds right to me, but would love feedback from others

Reply:
I've been thinking that adding a few diagrams of allowed/disallowed topologies to illustrate these rules might be helpful.